keps/sig-storage/3314-csi-volume-snapshot-delta/README.md

```diff
@@ -0,0 +1,1484 @@
+the `cbt-aggapi` aggregated API server.
+
+In setup where there may be multiple replicas of the CSI driver, an active/
+passive leader election process will be used to elect a single leader instance
+of the `cbt-http` sidecar, while idling other non-leader instances. Non-leader
+instances will voluntarily fail their readiness probe to remove themselves from
+the `Service`'s request path.
+
+### API Specification
+
+#### VolumeSnapshotDelta Resource
+
+The section describes the specification of `VolumeSnapshotDelta` resource:
+
+```go
+// VolumeSnapshotDelta represents a VolumeSnapshotDelta resource.
+type VolumeSnapshotDelta struct {
+  metav1.TypeMeta `json:",inline"`
+
+  // Standard object's metadata.
+  // More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+  // +optional
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+
+  // spec defines the desired characteristics of a snapshot delta requested by a user.
+  // Required.
+  Spec VolumeSnapshotDeltaSpec `json:"spec"`
+
+  // status represents the current information of a snapshot delta.
+  // +optional
+  Status VolumeSnapshotDeltaStatus `json:"status,omitempty"`
+}
+
+// VolumeSnapshotDeltaSpec is the spec of a VolumeSnapshotDelta resource.
+type VolumeSnapshotDeltaSpec struct {
+  // The name of the base CSI volume snapshot to use for comparison.
+  // If not specified, return all changed blocks.
+  // +optional
+  FromVolumeSnapshotName string `json:"fromVolumeSnapshotName,omitempty"`
+
+  // The name of the target CSI volume snapshot to use for comparison.
+  // Required.
+  ToVolumeSnapshotName string `json:"toVolumeSnapshotName"`
+
+  // VolumeSnapshotDeltaRef is a reference to the secret object containing
+  // sensitive information to pass to the CSI driver to complete the CSI
+  // calls for VolumeSnapshotDelta.
+  // This field is optional, and may be empty if no secret is required. If the
+  // secret object contains more than one secret, all secrets are passed.
+  // +optional
+  VolumeSnapshotDeltaSecretRef *SecretReference
+
+  // Define the maximum number of entries to return in the response.
+  Limit uint64 `json:"limit"`
+
+  // Defines the start of the block index (in bytes).
+  OffsetBytes uint64 `json:"offsetBytes"`
+}
+
+// VolumeSnapshotDeltaStatus is the status for a VolumeSnapshotDelta resource
+type VolumeSnapshotDeltaStatus struct {
+  // The list of CBT entries.
+  ChangedBlocks []*ChangedBlocks `json:"changedBlocks;omitempty"`
+
+  // Captures any error encountered.
+  Error string `json:"error,omitempty"`
+
+  // A very brief description to communicate the current state of the CBT
+  // operation.
+  State VolumeSnapshotDeltaState `json:"state"`
+
+  // The limit defined in the request.
+  Limit uint64 `json:"limit"`
+
+  // The offset (in bytes) defined in the request.
+  OffsetBytes uint64 `json:"offsetBytes"`
+
+  // The starting block index of the next request.
+  Continue uint64 `json:"continue"`
```

Comment:
Rename to "Next"?

Reply:
`continue` is the [convention](https://kubernetes.io/docs/reference/using-api/api-concepts/#retrieving-large-results-sets-in-chunks) used in the K8s API for chunk data.